<launch>
  <!-- <arg name="x_car1" default="-12"/>
  <arg name="y_car1" default="-10"/>
  <arg name="z_car1" default="0"/> -->
  <arg name="x_car1" default="-27.661731409782014"/>
  <arg name="y_car1" default="-30"/>
  <arg name="z_car1" default="0"/>
  <arg name="roll_car1" default="0"/>
  <arg name="pitch_car1" default="0"/>
  <arg name="yaw_car1" default="0.78539816339744828"/>
  <arg name="x_car2" default="-31.169134295108993"/>
  <arg name="y_car2" default="-27.975000000000001"/>
  <arg name="z_car2" default="0"/>
  <arg name="roll_car2" default="0"/>
  <arg name="pitch_car2" default="0"/>
  <arg name="yaw_car2" default="0.78539816339744828"/>
  <arg name="x_diff1" default="-31.169134295108993"/>
  <arg name="y_diff1" default="-32.024999999999999"/>
  <arg name="yaw_diff1" default="0.78539816339744828"/>
  <arg name="robot1_name"  default="mir1"/>
  <arg name="robot2_name"  default="mir2"/>
  <arg name="robot3_name"  default="mir3"/>
  <arg name="robot4_name"  default="mir4"/>
  <arg name="robot5_name"  default="mir5"/>

  <!-- <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="-5.25"/>
  <arg name="y_pos" default="-1.5"/>
  <arg name="YAW" default="0"/>

  <arg name="model1" default="$(env TURTLEBOT3_MODEL)" doc="model1 type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos1" default="-6.45"/>
  <arg name="y_pos1" default="-1.5"/>
  <arg name="YAW1" default="0"/> -->

  <arg name="namespace" default="/"/>
  <arg name="cmd_timeout" default="0.5"/>

  <env name="GAZEBO_MODEL_PATH" value="${GAZEBO_MODEL_PATH}:$(find gazebo_sfm_plugin)/media/models"/>
  <env name="GAZEBO_RESOURCE_PATH" value="${GAZEBO_RESOURCE_PATH}:$(find gazebo_sfm_plugin)/media/models"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find gazebo_sfm_plugin)/worlds/cafe3.world"/>  
    <!-- <arg name="paused" value="false"/> -->
    <!-- <arg name="use_sim_time" value="true"/> -->
    <!-- <arg name="gui" value="true"/> -->
    <!-- <arg name="headless" value="false"/> -->
    <!-- <arg name="debug" value="true"/> -->
  </include>
  <group ns="$(arg robot1_name)">
    <include file="$(find mir_launch_sim)/launch/mir_100.launch">
        <arg name="robot_x"     default="-3.2786977291107178" />
        <arg name="robot_y"     default="-1.0815595388412476" />
        <arg name="robot_yaw"   default="0.68417871581859846" /> 
        <arg name="tf_prefix"   default="$(arg robot1_name)" />       
        <arg name="model_name"  default="$(arg robot1_name)"/>
    </include>
  </group>
  <group ns="$(arg robot2_name)">
    <include file="$(find mir_launch_sim)/launch/mir_100.launch">
        <arg name="robot_x"     default="0.05000000074505806" />
        <arg name="robot_y"     default="-3.5" />
        <arg name="robot_yaw"   default="0.95138371283847112" /> 
        <arg name="tf_prefix"   default="$(arg robot2_name)" />       
        <arg name="model_name"  default="$(arg robot2_name)"/>
    </include>
  </group>
  <group ns="$(arg robot3_name)">
    <include file="$(find mir_launch_sim)/launch/mir_100.launch">
        <arg name="robot_x"     default="3.3786978721618652" />
        <arg name="robot_y"     default="-1.0815595388412476" />
        <arg name="robot_yaw"   default="1.0894137561692816" /> 
        <arg name="tf_prefix"   default="$(arg robot3_name)" />       
        <arg name="model_name"  default="$(arg robot3_name)"/>
    </include>
  </group>
  <group ns="$(arg robot4_name)">
    <include file="$(find mir_launch_sim)/launch/mir_100.launch">
        <arg name="robot_x"     default="-2.0151755809783936" />
        <arg name="robot_y"     default="2.8257830142974854" />
        <arg name="robot_yaw"   default="1.1139706296366882" /> 
        <arg name="tf_prefix"   default="$(arg robot4_name)" />       
        <arg name="model_name"  default="$(arg robot4_name)"/>
    </include>
  </group>

  <include file="/home/weijian/VisFormation/src/vis_formation_planner/launch/traj_tracking/control_triangular_formation.launch">
  </include>

  <!-- <include file="$(find racebot_control)/launch/racebot_control.launch"/> -->
  <!-- <node pkg="rviz" type="rviz" name="rviz" args="" />  -->
  <!--car-like robot 1-->
  <!-- <include file="$(find hunter2_control)/launch/hunter2_contol.launch" ns="/"/> -->

  <!-- urdf xml robot description loaded on the Parameter Server, converting the xacro into a proper urdf file-->
  <!-- <param name="robot_description" command="$(find xacro)/xacro '$(find hunter2_gazebo)/xacro/hunter2_base.xacro'" /> -->
  <!-- <group ns = "hunter2_base1">
    <include file="$(find hunter2_control)/launch/hunter2_v_contol.launch" />
    <param name="robot_description" command="$(find xacro)/xacro '$(find hunter2_base)/urdf/hunter2_base_gazebo.xacro' bodyframe:=hunter2_base1 ns:=hunter2_base1" />
    <node name="hunter_spawn" pkg="gazebo_ros" type="spawn_model" output="screen" 
    args="-urdf -param robot_description -model hunter2_base1 -x $(arg x_car1) -y $(arg y_car1) -z 1.5 -R $(arg roll_car1) -P $(arg pitch_car1) -Y $(arg yaw_car1)"
        respawn="false" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
        <param name="publish_frequency" value="10.0"/>
        <param name="tf_prefix" value="hunter2_base1"/>
    </node>
  </group>

  <group ns = "hunter2_base2">
    <include file="$(find hunter2_control)/launch/hunter2_v_contol.launch" />
    <param name="robot_description" command="$(find xacro)/xacro '$(find hunter2_base)/urdf/hunter2_base_gazebo.xacro' bodyframe:=hunter2_base2 ns:=hunter2_base2" />
    <node name="hunter_spawn" pkg="gazebo_ros" type="spawn_model" output="screen" 
    args="-urdf -param robot_description -model hunter2_base2 -x $(arg x_car2) -y $(arg y_car2) -z 1.5 -R $(arg roll_car2) -P $(arg pitch_car2) -Y $(arg yaw_car2)"
        respawn="false" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
        <param name="publish_frequency" value="10.0"/>
        <param name="tf_prefix" value="hunter2_base2"/>
    </node>
  </group> -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d /home/weijian/VisFormation/src/vis_formation_planner/launch/rviz_config/single_mir_visualization.rviz" />
</launch>